aimmdb/ingest/eli.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+import pandas as pd
+from pathlib import Path
+
+
+def read_metadata_and_header(path):
+    """Read through commented lines of dat file to get metadata and DataFrame header
+
+    Parameters
+    ----------
+    path: str, or path object
+        path to .dat file from ISS beamline
+
+    Returns
+    -------
+    tuple[dict[str, str], str]
+        First element is dictionary of metadata.
+        Second element is header as single string, with column names separated by whitespace.
+    """
+    metadata: dict
+    header: str
+    with open(path, "r") as dat_file:
+        lines = dat_file.readlines()
```

Comment:
This is the only line that needs to be under the `with` context manager. You should de-indent those which come after but are still under that context.